Builder.cs

```diff
@@ -291,40 +291,40 @@ private void WriteItems()
             }
 
             //write free item (start item default = Deku Mask)
-            var freeItemIndex = _randomized.ItemList.FindIndex(u => u.ReplacesItemId == Items.MaskDeku);
-            WriteFreeItem(_randomized.ItemList[freeItemIndex].ID);
+            var freeItem = _randomized.ItemList.Find(u => u.ReplacesItemId == Items.MaskDeku);
+            WriteFreeItem(freeItem.ID);
 
             //write everything else
             ItemSwapUtils.ReplaceGetItemTable(Values.ModsDirectory);
             ItemSwapUtils.InitItems();
 
-            for (int i = 0; i < _randomized.ItemList.Count; i++)
+            foreach (var item in _randomized.ItemList)
             {
-                var itemId = _randomized.ItemList[i].ID;
-
                 // Unused item
-                if (_randomized.ItemList[i].ReplacesItemId == -1)
+                if (item.ReplacesItemId == -1)
```

Comment:
you could also use `if(!item.ReplacesAnotherItem)` if you wanted. its still fairly readable as is imo